server/src/main/java/io/druid/server/coordination/ZkCoordinator.java

```diff
@@ -256,8 +256,14 @@ public void loadLocalCache()
       File file = segmentsToLoad[i];
       log.info("Loading segment cache file [%d/%d][%s].", i, segmentsToLoad.length, file);
       try {
-        DataSegment segment = jsonMapper.readValue(file, DataSegment.class);
-        if (serverManager.isSegmentCached(segment)) {
+        final DataSegment segment = jsonMapper.readValue(file, DataSegment.class);
+
+        if (!segment.getIdentifier().equals(file.getName())) {
```

Comment:
I feel like it's better to keep it around and let the user delete it on their own when they're ready, in case they want to use the file to reindex or perform some other corrective action.